tensorflow/python/ops/math_ops.py

```diff
@@ -763,99 +762,126 @@ def saturate_cast(value, dtype, name=None):
   # When casting to a type with smaller representable range, clamp.
   # Note that this covers casting to unsigned types as well.
   with ops.name_scope(name, "saturate_cast", [value]) as name:
-    value = ops.convert_to_tensor(value, name="value")
     dtype = dtypes.as_dtype(dtype).base_dtype
-    if value.dtype.min < dtype.min:
-      value = gen_math_ops.maximum(value,
-                                   ops.convert_to_tensor(
-                                       dtype.min, dtype=value.dtype,
-                                       name="min"))
-    if value.dtype.max > dtype.max:
-      value = gen_math_ops.minimum(value,
-                                   ops.convert_to_tensor(
-                                       dtype.max, dtype=value.dtype,
-                                       name="max"))
-    return cast(value, dtype, name=name)
-
-
-def to_float(x, name="ToFloat"):
+    if isinstance(value, sparse_tensor.SparseTensor):
+      #SparseTensor execute
+      values_cast = saturate_cast(value.values, dtype, name=name)
+      return sparse_tensor.SparseTensor(value.indices, values_cast, value.dense_shape)
+    else:
+      value = ops.convert_to_tensor(value, name="value")
+      if value.dtype.min < dtype.min:
+        value = gen_math_ops.maximum(value,
+                                     ops.convert_to_tensor(
+                                         dtype.min, dtype=value.dtype,
+                                         name="min"))
+      if value.dtype.max > dtype.max:
+        value = gen_math_ops.minimum(value,
+                                     ops.convert_to_tensor(
+                                         dtype.max, dtype=value.dtype,
+                                         name="max"))
+      return cast(value, dtype, name=name)
+
+
+def to_float(x, name="ToFloat", mode="unsafe"):
   """Casts a tensor to type `float32`.
 
   Args:
     x: A `Tensor` or `SparseTensor`.
     name: A name for the operation (optional).
+    mode: "safe"/"unsafe" mode setting
 
   Returns:
     A `Tensor` or `SparseTensor` with same shape as `x` with type `float32`.
 
   Raises:
     TypeError: If `x` cannot be cast to the `float32`.
   """
-  return cast(x, dtypes.float32, name=name)
+  if mode == "safe":
+    return saturate_cast(x, dtypes.float32, name=name)
+  else:
+    return cast(x, dtypes.float32, name=name)
 
 
-def to_double(x, name="ToDouble"):
+def to_double(x, name="ToDouble", mode="unsafe"):
   """Casts a tensor to type `float64`.
 
   Args:
     x: A `Tensor` or `SparseTensor`.
     name: A name for the operation (optional).
+    mode: "safe"/"unsafe" mode setting
 
   Returns:
     A `Tensor` or `SparseTensor` with same shape as `x` with type `float64`.
 
   Raises:
     TypeError: If `x` cannot be cast to the `float64`.
   """
-  return cast(x, dtypes.float64, name=name)
+  if mode == "safe":
+    return saturate_cast(x,dtypes.float64, name=name)
```

Comment:
nit: space after x; more generally please review your python code throughout for spacing and other small nits.